app/models/tenant.rb

```diff
@@ -57,8 +58,8 @@ class Tenant < ApplicationRecord
   virtual_column :display_type, :type => :string
 
   before_save :nil_blanks
-  after_create :create_tenant_group
-  before_destroy :ensure_can_be_destroyed
+  after_create :create_tenant_group, :create_miq_product_features_for_tenant_nodes
+  before_destroy :ensure_can_be_destroyed, :destroy_tenant_feature_for_tenant_node
```

Comment:
I don't think that you need to call `destroy_tenant_feature_for_tenant_node`. The `has_many :miq_product_features, :dependent => :destroy` in this class should take care of deleting the product features associated with the tenant being deleted.

Reply:
you are right, thanks 👍